### ISSUE TITLE
Using flow spec to block protocol ranges 

### ISSUE THREAD

**Issue by GlassRod** [Exa-Networks/exabgp#588]
##### ISSUE TYPE
 - Feature Idea

##### OS
ubuntu

##### VERSION
master

##### ENVIRONMENT 
N/A

##### CONFIGURATION
N/A

##### SUMMARY
we wish to use flow spec to block a whole protocol range in case of attack. The flow spec RFC supports that but exabgp allows input of only a single protocol value. is there a way to block a protocol range say 1-47 ? if not is there a plan to add this ?

##### STEPS TO REPRODUCE
(missing)

##### EXPECTED RESULTS
(missing)

##### ACTUAL RESULTS
N/A

##### IMPORTANCE
missing

**Comment by thomas-mangin** [comment]
Can you please provide me with an example of flow as you would like it to work - as I do not have your use case I can not see exactly what you need.

**Comment by YuriyAr** [comment]
Hi,
I'm speaking in the name of GlassRod for this case.
Let me explain by example as you requested - attached.
Basically this requires 2 changes: use protocol numbers instead of names, and support ranges in this field (similarly to what is supported in, e.g., source port field).

Thanks, Yuriy.

[CurrentConfig.txt](https://github.com/Exa-Networks/exabgp/files/853083/CurrentConfig.txt)
[DesiredConfig.txt](https://github.com/Exa-Networks/exabgp/files/853082/DesiredConfig.txt)

**Comment by thomas-mangin** [comment]
```
  neighbor 11.11.11.11 {
    description "11.11.11.11";
    router-id 10.10.10.10;
    local-address 10.10.10.10;
    local-as 65001;
    peer-as 65001;
    hold-time 180;
    flow {
      route Flow1489910466608_0 {
        match {
          source 80-81;
          destination 12.12.12.12/32;
          protocol  [ >10&<13 17 ];
        }
        then {
          discard;
        }
      }
    }
  }
```

Seems to work on master but not 3.4

**Comment by YuriyAr** [comment]
Thanks a lot!
Is master branch recommended for production deployments?
If not, is stable branch planned in the foreseeable future?
Thanks again, Yuriy.

**Comment by thomas-mangin** [comment]
3.4 is still the recommended stable version but 4.0.0 is around the corner (I had many demand for it)
The only thing to know is that the template format will change slightly before version 4.
Most likely going to do this change this week.

**Comment by YuriyAr** [comment]
This is great, thanks!
I would appreciate if you could provide a planned date for 4.0.0 release, even if it is tentative yet.
Yuriy.

**Comment by thomas-mangin** [comment]
I am one year out on my last guess, but I am working with some people who have deadline so I would guess less than a month.

**Comment by thomas-mangin** [comment]
If you can help fixing python3 bugs on the py3 branch .. this is the only thing holding the release.

**Comment by YuriyAr** [comment]
Great, thanks, this is very useful information!
I will check about helping fix py3 bugs.
Yuriy.

**Comment by GlassRod** [comment]
Any news about 4.0.0 ETA ?

**Comment by thomas-mangin** [comment]
AFAIK this has now been fixed and works on master (perhaps even 4.0.0)
